### PR TITLE
Raise ValueError when calling `xxx_obj` with ChainerX array in ChianerMN

### PR DESCRIPTION
**Pull request by belldandyxtq**
The chainerx array preserves device id inside, which will also be
pickled during the mpi communications and causes future errors.

This PR prevents such cases by raising ValueError when calling
`x_obj` with the chainerx array.

### CODE REVIEW

**Comment by shu65** [comment]
Jenkins, test this please

File: chainermn/communicators/mpi_communicator_base.py [hunk 1]
```diff
@@ -617,23 +618,77 @@ def scatter(self, xs, root=0):
...
+        elif type(obj) is dict:
+            for key, value in obj.items():
+                xp = chainer.backend.get_array_module(key)
+                if xp == chainerx and key.device.name.startswith('cuda'):
+                    return True
+
+                xp = chainer.backend.get_array_module(value)
+                if xp == chainerx and value.device.name.startswith('cuda'):
+                    return True
+        else:
```
Comment (shu65) [inline chainermn/communicators/mpi_communicator_base.py]: How about checking if `obj` is array object of `numpy`, 'cupy', and `chainerx`?

Comment (belldandyxtq) [reply chainermn/communicators/mpi_communicator_base.py]: `chainer.backend.get_array_module` can accept any type and return `numpy` for types that are not cupy nor chainerx. So I don't think we have to check if they beforehand.

**Comment by belldandyxtq** [comment]
Jenkins, test this please

**Pull request merged**